framework/integration/testresource/wrapper.go

```diff
@@ -0,0 +1,47 @@
+// +build k8srequired
+
+package testresource
+
+import (
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	createCount = 0
+	deleteCount = 0
+)
+
+type Config struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+	Name      string
+}
+
+type Wrapper struct {
+	Resource framework.Resource
+}
+
+func New(config Config) (*Wrapper, error) {
+	r := &Resource{
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+
+		name: config.Name,
+	}
+
+	tr := &Wrapper{
+		Resource: r,
+	}
+
+	return tr, nil
+}
+
+func (r *Wrapper) GetCreateCount() int {
+	return createCount
+}
+
+func (r *Wrapper) GetDeleteCount() int {
```

Comment:
In case you would like to wrap a resource that way you should wrap the whole resource implementation. This here makes no sense somehow. Like why does the wrapper have the resource at all? Lets maybe just implement the test resource properly and that is it.

Reply:
I believe I fixed this.